File: php/testcode/codeinj_035.php
<?php
require_once __DIR__ . '/shared.php';

// vuln-code-snippet start php_codeinj_hardcoded_comparator
function codeinj035(BenchmarkRequest $req): BenchmarkResponse {
    $items = explode(',', $req->param('items'));
    usort($items, fn($a, $b) => strcmp($a, $b)); // vuln-code-snippet safe-line php_codeinj_hardcoded_comparator
    return BenchmarkResponse::ok(implode(',', $items));
}
// vuln-code-snippet end php_codeinj_hardcoded_comparator

File: ruby/testcode/codeinj_035.rb
require_relative 'shared'

# vuln-code-snippet start ruby_codeinj_catch_throw_eval
def evaluate_and_catch(req)
  expr = req.param('expr')
  result = catch(:result) { throw :result, eval(expr) } # vuln-code-snippet vuln-line ruby_codeinj_catch_throw_eval
  BenchmarkResponse.json({ result: result.to_s })
end
# vuln-code-snippet end ruby_codeinj_catch_throw_eval
